### ISSUE TITLE
curl: (35) Unknown SSL protocol error in connection to sh.rustup.rs:443

### ISSUE THREAD

**Issue by pierrenel** [rust-lang/rustup#1274]
I've seen other issues here regarding this err... issue - it seems to be happening again: I'm on High Sierra.

```curl 7.43.0 (x86_64-apple-darwin10.8.0) libcurl/7.43.0 OpenSSL/0.9.8zd zlib/1.2.8 libidn/1.17```

**Comment by AsceticBoy** [comment]
Same here 😩 @pierrenel how to solve it

**Comment by ChenYizhu97** [comment]
have you solved it?

**Comment by tyler2018** [comment]
same here. 😩

**Comment by kinnison** [comment]
Is anyone still experiencing this?

**Comment by 0ex-d** [comment]
I am using a macOS Sierra is that really old?

**Comment by rbtcollins** [comment]
Yes, High Sierra replaced it in September 2017 https://en.wikipedia.org/wiki/MacOS_Sierra
Followed by Mojave and Catalina. SSL protocol issues to keep up with security breaches are a constant moving target unfortunately.

**Comment by y3g0r** [comment]
Had the same issue on MacOS Catalina 10.15.7 (19H1030)

`curl 7.54.0 (x86_64-apple-darwin16.5.0) libcurl/7.54.0 SecureTransport zlib/1.2.11`

Installing more recent version of curl using brew helped

```sh
brew install curl
export PATH="/usr/local/opt/curl/bin:$PATH"
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
